xml-format goal fails to execute on maven versions prior to 3.3.9

My team is seeing this plugin fail on maven versions prior to 3.3.9. I've tried with 3.0.5 and 3.3.3 and both receive this error: 
`[ERROR] Failed to execute goal au.com.acegi:xml-format-maven-plugin:3.0.6:xml-format (default) on project [name redacted]: Unable to parse configuration of mojo au.com.acegi:xml-format-maven-plugin:3.0.6:xml-format for parameter includes: Cannot assign configuration entry 'includes' with value '**/src/test/resources/**/*.xml' of type java.lang.String to property of type java.lang.String[] -> [Help 1]`

Here's how we have the plugin configured in our pom. 
```xml
<plugin>
    <groupId>au.com.acegi</groupId>
    <artifactId>xml-format-maven-plugin</artifactId>
    <version>3.0.6</version>
    <executions>
        <execution>
            <goals>
                <goal>xml-format</goal>
            </goals>
        </execution>
    </executions>
    <configuration>
        <indentSize>4</indentSize>
        <includes>**/src/test/resources/**/*.xml</includes>
    </configuration>
</plugin>
```

Maven versions from 3.3.9 onward seem to parse this correctly. Are there any other alternative ways of writing the `includes` tag? We've tried:
```xml
<includes>
    <include>**/src/test/resources/**/*.xml</include>
</includes>
```
We've also tried: 

```xml
<includes>
    <includes>**/src/test/resources/**/*.xml</includes>
</includes>
```
But neither of these formats gave us the results we're looking for as it formatted all XML files in the project and not just the ones in the folder we would like to specify. 

The [official documentation](https://acegi.github.io/xml-format-maven-plugin/plugin-info.html) states that only a minimum version of 2.0 is required for maven.

We have a large number of engineers and build machines that make requiring a minimum maven version of 3.3.9 infeasible at this time. However, we would still like to be able to use this plugin. Any guidance would be appreciated.

Thanks for the suggestions @benalexau. Here are the results. 

With the following configuration I do not get a build failure. _However_, xml files outside of `/src/test/resources` are being formatted so it's still not the desired result.
```xml
<includes>
    <include>src/test/resources/**/*.xml</include>
</includes>
```

And below is the debug information for the original failure. I apologise as I should have included this from the start. 
```
[INFO] --- xml-format-maven-plugin:3.0.6:xml-format (default) @ com.cerner.registration.conversation.type ---
[DEBUG] Dependency collection stats: {ConflictMarker.analyzeTime=0, ConflictMarker.markTime=0, ConflictMarker.nodeCount=11, ConflictIdSorter.graphTime=0, ConflictIdSorter.topsortTime=0, ConflictIdSorter.conflictIdCount=8, ConflictIdSorter.conflictIdCycleCount=0, ConflictResolver.totalTime=0, ConflictResolver.conflictItemCount=11, DefaultDependencyCollector.collectTime=50, DefaultDependencyCollector.transformTime=0}
[DEBUG] au.com.acegi:xml-format-maven-plugin:jar:3.0.6:
[DEBUG]    org.apache.maven:maven-plugin-api:jar:3.5.0:compile
[DEBUG]       org.apache.maven:maven-model:jar:3.5.0:compile
[DEBUG]          org.apache.commons:commons-lang3:jar:3.5:compile
[DEBUG]       org.apache.maven:maven-artifact:jar:3.5.0:compile
[DEBUG]    org.codehaus.plexus:plexus-utils:jar:3.1.0:compile
[DEBUG]    org.dom4j:dom4j:jar:2.0.1:compile
[DEBUG]       jaxen:jaxen:jar:1.1.6:compile
[DEBUG] Created new class realm plugin>au.com.acegi:xml-format-maven-plugin:3.0.6
[DEBUG] Importing foreign packages into class realm plugin>au.com.acegi:xml-format-maven-plugin:3.0.6
[DEBUG]   Imported:  < project>com.cerner.registration:com.cerner.registration.conversation.type:1.0-SNAPSHOT
[DEBUG] Populating class realm plugin>au.com.acegi:xml-format-maven-plugin:3.0.6
[DEBUG]   Included: au.com.acegi:xml-format-maven-plugin:jar:3.0.6
[DEBUG]   Included: org.apache.commons:commons-lang3:jar:3.5
[DEBUG]   Included: org.codehaus.plexus:plexus-utils:jar:3.1.0
[DEBUG]   Included: org.dom4j:dom4j:jar:2.0.1
[DEBUG]   Included: jaxen:jaxen:jar:1.1.6
[DEBUG] Configuring mojo au.com.acegi:xml-format-maven-plugin:3.0.6:xml-format from plugin realm ClassRealm[plugin>au.com.acegi:xml-format-maven-plugin:3.0.6, parent: sun.misc.Launcher$AppClassLoader@15d56d5]
[DEBUG] Configuring mojo 'au.com.acegi:xml-format-maven-plugin:3.0.6:xml-format' with basic configurator -->
[DEBUG]   (f) attributeQuoteChar = "
[DEBUG]   (f) baseDirectory = <project directory>
[DEBUG]   (f) encoding = UTF-8
[DEBUG]   (f) excludes = []
[DEBUG]   (f) expandEmptyElements = false
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 38.364 s
[INFO] Finished at: 2018-02-05T08:10:09-06:00
[INFO] Final Memory: 49M/129M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal au.com.acegi:xml-format-maven-plugin:3.0.6:xml-format (default) on project com.cerner.registration.conversation.type: Unable to parse configuration of mojo au.com.acegi:xml-format-maven-plugin:3.0.6:xml-format for parameter includes: Cannot assign configuration entry 'includes' with value 'src/test/resources/**/*.xml' of type java.lang.String to property of type java.lang.String[] -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal au.com.acegi:xml-format-maven-plugin:3.0.6:xml-format (default) on project com.cerner.registration.conversation.type: Unable to parse configuration of mojo au.com.acegi:xml-format-maven-plugin:3.0.6:xml-format for parameter includes: Cannot assign configuration entry 'includes' with value 'src/test/resources/**/*.xml' of type java.lang.String to property of type java.lang.String[]
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:220)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:153)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:145)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:116)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:80)
	at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:51)
	at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:128)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:307)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:193)
	at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:106)
	at org.apache.maven.cli.MavenCli.execute(MavenCli.java:862)
	at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:286)
	at org.apache.maven.cli.MavenCli.main(MavenCli.java:197)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:606)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:289)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:229)
	at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:415)
	at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:356)
Caused by: org.apache.maven.plugin.PluginConfigurationException: Unable to parse configuration of mojo au.com.acegi:xml-format-maven-plugin:3.0.6:xml-format for parameter includes: Cannot assign configuration entry 'includes' with value 'src/test/resources/**/*.xml' of type java.lang.String to property of type java.lang.String[]
	at org.apache.maven.plugin.internal.DefaultMavenPluginManager.populatePluginFields(DefaultMavenPluginManager.java:689)
	at org.apache.maven.plugin.internal.DefaultMavenPluginManager.getConfiguredMojo(DefaultMavenPluginManager.java:621)
	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:121)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:208)
	... 20 more
Caused by: org.codehaus.plexus.component.configurator.ComponentConfigurationException: Cannot assign configuration entry 'includes' with value 'src/test/resources/**/*.xml' of type java.lang.String to property of type java.lang.String[]
	at org.codehaus.plexus.component.configurator.converters.AbstractConfigurationConverter.failIfNotTypeCompatible(AbstractConfigurationConverter.java:100)
	at org.codehaus.plexus.component.configurator.converters.composite.ArrayConverter.fromConfiguration(ArrayConverter.java:65)
	at org.eclipse.sisu.plexus.CompositeBeanHelper.convertProperty(CompositeBeanHelper.java:273)
	at org.eclipse.sisu.plexus.CompositeBeanHelper.setProperty(CompositeBeanHelper.java:210)
	at org.codehaus.plexus.component.configurator.converters.composite.ObjectWithFieldsConverter.processConfiguration(ObjectWithFieldsConverter.java:101)
	at org.codehaus.plexus.component.configurator.BasicComponentConfigurator.configureComponent(BasicComponentConfigurator.java:34)
	at org.apache.maven.plugin.internal.DefaultMavenPluginManager.populatePluginFields(DefaultMavenPluginManager.java:659)
	... 23 more
[ERROR] 
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/PluginConfigurationException
```

Again, maven versions 3.3.9 and greater are not failing with the configuration specified like this:
```xml
<includes>src/test/resources/**/*.xml</includes>
```

--- a/src/main/java/au/com/acegi/xmlformat/XmlFormatPlugin.java
+++ b/src/main/java/au/com/acegi/xmlformat/XmlFormatPlugin.java
@@ -192,9 +192,7 @@ public final class XmlFormatPlugin extends AbstractMojo {
       return;
     }
 
-    if (includes == null || includes.length == 0) {
-      includes = new String[]{"**/*.xml"};
-    }
+    initializeIncludes();
 
     final OutputFormat fmt = buildFormatter();
 
@@ -239,6 +237,12 @@ public final class XmlFormatPlugin extends AbstractMojo {
 
   void setTargetDirectory(final File targetDirectory) {
     this.targetDirectory = targetDirectory;
+  }
+  
+  private void initializeIncludes() {
+    if (includes == null || includes.length == 0) {
+      includes = new String[]{"**/*.xml"};
+    }
   }
 
   private OutputFormat buildFormatter() {

--- a/src/main/java/au/com/acegi/xmlformat/XmlFormatPlugin.java
+++ b/src/main/java/au/com/acegi/xmlformat/XmlFormatPlugin.java
@@ -70,7 +70,7 @@ public final class XmlFormatPlugin extends AbstractMojo {
    * the formatting. In addition to these exclusions, the project build
    * directory (typically <code>target</code>) is always excluded.
    */
-  @Parameter(property = "excludes", defaultValue = "")
+  @Parameter(property = "excludes")
   private String[] excludes;
 
   /**
@@ -185,7 +185,6 @@ public final class XmlFormatPlugin extends AbstractMojo {
   public void execute() throws MojoExecutionException, MojoFailureException {
     assert baseDirectory != null;
     assert targetDirectory != null;
-    assert excludes != null;
 
     if (skip) {
       getLog().info("[xml-format] Skipped");
@@ -193,6 +192,7 @@ public final class XmlFormatPlugin extends AbstractMojo {
     }
 
     initializeIncludes();
+    initializeExcludes();
 
     final OutputFormat fmt = buildFormatter();
 
@@ -242,6 +242,12 @@ public final class XmlFormatPlugin extends AbstractMojo {
   private void initializeIncludes() {
     if (includes == null || includes.length == 0) {
       includes = new String[]{"**/*.xml"};
+    }
+  }
+  
+  private void initializeExcludes() {
+    if (excludes == null || excludes.length == 0) {
+      excludes = new String[0];
     }
   }
 

--- a/src/main/java/au/com/acegi/xmlformat/XmlFormatPlugin.java
+++ b/src/main/java/au/com/acegi/xmlformat/XmlFormatPlugin.java
@@ -82,7 +82,7 @@ public final class XmlFormatPlugin extends AbstractMojo {
    * A set of file patterns that dictate which files should be included in the
    * formatting with each file pattern being relative to the base directory.
    */
-  @Parameter(property = "includes", defaultValue = "**/*.xml")
+  @Parameter(property = "includes")
   private String[] includes;
 
   /**
@@ -185,12 +185,15 @@ public final class XmlFormatPlugin extends AbstractMojo {
   public void execute() throws MojoExecutionException, MojoFailureException {
     assert baseDirectory != null;
     assert targetDirectory != null;
-    assert includes != null && includes.length > 0;
     assert excludes != null;
 
     if (skip) {
       getLog().info("[xml-format] Skipped");
       return;
+    }
+
+    if (includes == null || includes.length == 0) {
+      includes = new String[]{"**/*.xml"};
     }
 
     final OutputFormat fmt = buildFormatter();

--- a/src/main/java/au/com/acegi/xmlformat/XmlFormatPlugin.java
+++ b/src/main/java/au/com/acegi/xmlformat/XmlFormatPlugin.java
@@ -193,7 +193,7 @@ public final class XmlFormatPlugin extends AbstractMojo {
     }
 
     if (includes == null || includes.length == 0) {
-        includes = new String[]{ "**/*.xml" };
+      includes = new String[]{"**/*.xml"};
     }
 
     final OutputFormat fmt = buildFormatter();

